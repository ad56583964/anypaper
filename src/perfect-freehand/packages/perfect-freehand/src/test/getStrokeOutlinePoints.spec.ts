import { getStrokePoints } from '../getStrokePoints'
import { getStrokeOutlinePoints } from '../getStrokeOutlinePoints'
import inputs from './inputs.json'

const { onePoint, twoPoints, twoEqualPoints, manyPoints, withDuplicates } =
  inputs

describe('getStrokeOutlinePoints', () => {
  for (const [key, value] of Object.entries(inputs)) {
    it(`runs ${key} without generating NaN values`, () => {
      expect(
        getStrokeOutlinePoints(getStrokePoints(value)).find((t) =>
          JSON.stringify(t).includes('null')
        )
      ).toBeUndefined()
    })
  }

  it('gets stroke outline points with a single point', () => {
    expect(getStrokeOutlinePoints(getStrokePoints(onePoint))).toMatchSnapshot(
      'get-stroke-outline-points-one-point'
    )
  })

  it('gets stroke outline points with two points', () => {
    expect(getStrokeOutlinePoints(getStrokePoints(twoPoints))).toMatchSnapshot(
      'get-stroke-outline-points-two-points'
    )
  })

  it('gets stroke outline points with two equal points', () => {
    expect(
      getStrokeOutlinePoints(getStrokePoints(twoEqualPoints)).find((t) =>
        JSON.stringify(t).includes('null')
      )
    ).toBeUndefined()

    expect(
      getStrokeOutlinePoints(getStrokePoints(twoEqualPoints))
    ).toMatchSnapshot('get-stroke-outline-points-two-equal-points')
  })

  it('gets stroke outline points on a line with a many points', () => {
    expect(getStrokeOutlinePoints(getStrokePoints(manyPoints))).toMatchSnapshot(
      'get-stroke-outline-points-many-points'
    )
  })

  it('gets stroke outline points with duplicates', () => {
    expect(
      getStrokeOutlinePoints(getStrokePoints(withDuplicates))
    ).toMatchSnapshot('get-stroke-outline-points-duplicates')
  })
})
