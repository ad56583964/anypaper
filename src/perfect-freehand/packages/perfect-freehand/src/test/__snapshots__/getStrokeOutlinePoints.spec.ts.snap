// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStrokeOutlinePoints gets stroke outline points on a line with a many points: get-stroke-outline-points-many-points 1`] = `
Array [
  Array [
    -4.7870689199405465,
    -3.409507067976618,
  ],
  Array [
    0.5508495821598389,
    -10.853391824710275,
  ],
  Array [
    10.487263712280766,
    -25.70547175940249,
  ],
  Array [
    16.909302387744244,
    -34.79842438528664,
  ],
  Array [
    23.53504451646879,
    -44.502838180176255,
  ],
  Array [
    30.392699457210753,
    -54.98003015785006,
  ],
  Array [
    37.74339990102968,
    -66.54666595477848,
  ],
  Array [
    45.492278070157795,
    -78.55740883476514,
  ],
  Array [
    53.41943624372377,
    -91.07701326875332,
  ],
  Array [
    61.11663417290783,
    -103.13773016048455,
  ],
  Array [
    66.83916275239665,
    -112.8234188226388,
  ],
  Array [
    72.2686181495529,
    -121.89389447146331,
  ],
  Array [
    77.1070920654364,
    -130.19614306631112,
  ],
  Array [
    81.56386517750026,
    -137.7900260266369,
  ],
  Array [
    86.225083882178,
    -144.79556338527095,
  ],
  Array [
    94.22836246806321,
    -157.6734732272444,
  ],
  Array [
    101.85858473906248,
    -170.05569306449985,
  ],
  Array [
    108.23814675691673,
    -179.39482312230325,
  ],
  Array [
    113.11338626008067,
    -186.94785496465755,
  ],
  Array [
    118.62323256145517,
    -195.59381676477452,
  ],
  Array [
    119.93325669186845,
    -197.119222029652,
  ],
  Array [
    121.4856479506413,
    -198.42903103110712,
  ],
  Array [
    123.30639427690807,
    -199.32925381241918,
  ],
  Array [
    125.28967380863872,
    -199.76756938324158,
  ],
  Array [
    127.32021824626513,
    -199.7185028220039,
  ],
  Array [
    129.28001225166767,
    -199.1849058795507,
  ],
  Array [
    131.05515250517186,
    -198.1977912352483,
  ],
  Array [
    132.5424677718394,
    -196.81453003860292,
  ],
  Array [
    133.6555152184904,
    -195.11551749535585,
  ],
  Array [
    134.32960447522913,
    -193.19950029434835,
  ],
  Array [
    134.52555744278206,
    -191.1778374453267,
  ],
  Array [
    134.23198532546817,
    -189.1680280880634,
  ],
  Array [
    133.46595054853915,
    -187.28688243683317,
  ],
  Array [
    132.27197508922214,
    -185.64373276527067,
  ],
  Array [
    135.3238949673716,
    -174.60614425031852,
  ],
  Array [
    134.40796654502861,
    -156.5654560983854,
  ],
  Array [
    131.40265736852416,
    -141.3091951068142,
  ],
  Array [
    128.46533458044902,
    -127.02351557922663,
  ],
  Array [
    124.99589393006602,
    -112.06407928892905,
  ],
  Array [
    121.41174181056502,
    -96.26573617805788,
  ],
  Array [
    117.6604945096032,
    -79.01402032163087,
  ],
  Array [
    113.75485310869797,
    -61.49424948296114,
  ],
  Array [
    109.76414300596984,
    -43.80400620029065,
  ],
  Array [
    106.02491274195988,
    -26.880431802191975,
  ],
  Array [
    103.14655799826791,
    -11.367373620209019,
  ],
  Array [
    100.81906406209607,
    3.2092015092298887,
  ],
  Array [
    98.75189963948858,
    17.140026089225994,
  ],
  Array [
    97.15696861228545,
    30.03644961395648,
  ],
  Array [
    96.25181867630182,
    41.50113153346337,
  ],
  Array [
    95.53681323750074,
    52.05969003852795,
  ],
  Array [
    95.15198424587182,
    61.83366998223893,
  ],
  Array [
    95.13648775167786,
    76.51487184148942,
  ],
  Array [
    95.46900532851154,
    88.48534369509058,
  ],
  Array [
    95.84937355209802,
    97.50366709174861,
  ],
  Array [
    96.31956110411606,
    105.86899720252657,
  ],
  Array [
    96.42201088671735,
    107.41722242825804,
  ],
  Array [
    96.19339873023004,
    109.29002762522434,
  ],
  Array [
    95.52322444102498,
    111.05369688162311,
  ],
  Array [
    94.45043858027876,
    112.60572565740684,
  ],
  Array [
    93.03739151262258,
    113.85590996720519,
  ],
  Array [
    91.36620960002034,
    114.73158904019304,
  ],
  Array [
    89.53402201375263,
    115.18186836750472,
  ],
  Array [
    87.64731558241832,
    115.1805776934806,
  ],
  Array [
    85.81574577245304,
    114.7277920321556,
  ],
  Array [
    84.14576350726794,
    113.84982730744,
  ],
  Array [
    82.73442823457935,
    112.59771087038521,
  ],
  Array [
    81.66376682674725,
    111.04421578714333,
  ],
  Array [
    80.99600617509962,
    109.27963126494303,
  ],
  Array [
    80.76995656008484,
    107.40651503912324,
  ],
  Array [
    78.74460035174697,
    99.51102050737828,
  ],
  Array [
    74.36127774786802,
    85.64352758410783,
  ],
  Array [
    71.34682583138213,
    77.14068673706913,
  ],
  Array [
    67.91963640005311,
    66.68641534776448,
  ],
  Array [
    63.745203196747696,
    55.17644639601926,
  ],
  Array [
    59.13877102546526,
    43.40541226397938,
  ],
  Array [
    54.34836607737416,
    31.80340092980233,
  ],
  Array [
    49.26159908136444,
    20.164206653849508,
  ],
  Array [
    44.32938615038735,
    8.518469858959886,
  ],
  Array [
    39.688381992915126,
    -2.2387370265287476,
  ],
  Array [
    35.26372536257266,
    -11.926273297805608,
  ],
  Array [
    30.959687487414065,
    -21.235473627045142,
  ],
  Array [
    26.985316659740292,
    -29.791475184727044,
  ],
  Array [
    20.67178511289104,
    -42.525503240030126,
  ],
  Array [
    15.283350617750273,
    -52.48757162017931,
  ],
  Array [
    10.72283116470662,
    -59.405717004495735,
  ],
  Array [
    6.812014813329286,
    -65.54894112195734,
  ],
  Array [
    5.34663135357798,
    -67.70036269475662,
  ],
  Array [
    4.644934202531846,
    -70.20707061077566,
  ],
  Array [
    4.780391237135472,
    -72.80661188950066,
  ],
  Array [
    5.738820070108675,
    -75.22681387283684,
  ],
  Array [
    7.4198729496385205,
    -77.21428077425765,
  ],
  Array [
    9.64754319548337,
    -78.560924284544,
  ],
  Array [
    12.188593144855435,
    -79.12575047268956,
  ],
  Array [
    14.776974197283414,
    -78.84962188769585,
  ],
  Array [
    17.14168217331186,
    -77.76144926530257,
  ],
  Array [
    19.035131523628316,
    -75.975164566324,
  ],
  Array [
    20.25907760143256,
    -73.67779227027653,
  ],
  Array [
    20.685372929288096,
    -71.10986786300373,
  ],
  Array [
    20.269384273837417,
    -68.54025370781328,
  ],
  Array [
    19.054665757116553,
    -66.23798908506586,
  ],
  Array [
    17.168398728480714,
    -64.444121708502,
  ],
  Array [
    14.80807584375423,
    -63.3464699659829,
  ],
  Array [
    12.220823532032673,
    -63.05995828470223,
  ],
  Array [
    9.67752779332159,
    -63.61458451332184,
  ],
  Array [
    7.444472362828884,
    -64.95227913929813,
  ],
  Array [
    5.755458733596541,
    -66.93298518217429,
  ],
  Array [
    4.787327078461473,
    -69.3493221964056,
  ],
  Array [
    4.641441037278266,
    -71.94829905875098,
  ],
  Array [
    5.3330749170689735,
    -74.45780220565963,
  ],
  Array [
    6.789814468045898,
    -76.61508599520573,
  ],
  Array [
    8.859138674679858,
    -78.19428224515616,
  ],
  Array [
    11.32438874628773,
    -79.0300486912738,
  ],
  Array [
    13.927452349609373,
    -79.03488036602924,
  ],
  Array [
    16.395788038234947,
    -78.20827139180011,
  ],
  Array [
    18.470960413988006,
    -76.6367679463918,
  ],
  Array [
    18.469297239951885,
    -76.638516788245,
  ],
  Array [
    23.728589484320054,
    -67.79036491954464,
  ],
  Array [
    28.259019572960376,
    -59.554176394278954,
  ],
  Array [
    33.349491156408774,
    -48.421432621521866,
  ],
  Array [
    38.972942605456026,
    -35.143124095219285,
  ],
  Array [
    42.668762045200374,
    -26.517991096584762,
  ],
  Array [
    46.80984959023685,
    -17.107973593541775,
  ],
  Array [
    51.1689611775994,
    -7.2230772128426555,
  ],
  Array [
    55.84494610804276,
    3.479868234822921,
  ],
  Array [
    60.95718818232176,
    14.79040558671775,
  ],
  Array [
    66.57221785429451,
    26.44251132310081,
  ],
  Array [
    71.92161014209849,
    37.96133935187074,
  ],
  Array [
    77.32701691303012,
    49.698429447548904,
  ],
  Array [
    82.5462485833856,
    61.47869836729639,
  ],
  Array [
    86.42400534463177,
    71.5437013469928,
  ],
  Array [
    89.94324921389257,
    80.64128835799266,
  ],
  Array [
    94.50827564342275,
    95.23186430244567,
  ],
  Array [
    96.42201134382653,
    107.41643982549144,
  ],
  Array [
    96.19358646900167,
    109.28926787430932,
  ],
  Array [
    95.52358855007283,
    111.05300413931856,
  ],
  Array [
    94.45095789756785,
    112.60514018592802,
  ],
  Array [
    93.03803585540767,
    113.855465794182,
  ],
  Array [
    91.3669415190685,
    114.73131198098243,
  ],
  Array [
    89.5347989698944,
    115.18177452480103,
  ],
  Array [
    87.6480924189262,
    115.18067252142619,
  ],
  Array [
    85.81647733955272,
    114.72807001934582,
  ],
  Array [
    84.1464072862452,
    113.85027229724626,
  ],
  Array [
    82.7349468089698,
    112.5982969999791,
  ],
  Array [
    81.66413005698293,
    111.0449089906453,
  ],
  Array [
    80.99619295022218,
    109.28039125333297,
  ],
  Array [
    80.76995602471538,
    107.40729764184022,
  ],
  Array [
    81.05286020607011,
    103.10263176921475,
  ],
  Array [
    81.61400404437686,
    93.24889915299345,
  ],
  Array [
    82.21483023048896,
    82.44537131294317,
  ],
  Array [
    83.18846311016978,
    69.19457134140492,
  ],
  Array [
    83.9382435674237,
    61.054440839366976,
  ],
  Array [
    84.92894150491934,
    51.03422768160232,
  ],
  Array [
    86.25723284578287,
    40.458121146605315,
  ],
  Array [
    87.77336569598853,
    28.72105219278804,
  ],
  Array [
    89.80850430436428,
    15.7259250456917,
  ],
  Array [
    92.14854753691519,
    1.7841213992608547,
  ],
  Array [
    94.62857099106155,
    -12.889329060090944,
  ],
  Array [
    97.34902062800934,
    -28.711072963776186,
  ],
  Array [
    100.76920994881625,
    -45.71534016780618,
  ],
  Array [
    104.30159977698617,
    -63.42957404419856,
  ],
  Array [
    107.56983061732264,
    -80.95121388790268,
  ],
  Array [
    110.46673339753966,
    -98.32927554739545,
  ],
  Array [
    112.88571814320935,
    -114.3715607652393,
  ],
  Array [
    115.08569640625693,
    -129.16649477207864,
  ],
  Array [
    116.69461187584118,
    -143.26747028513412,
  ],
  Array [
    117.62272140476864,
    -157.55608277807076,
  ],
  Array [
    117.82643868218716,
    -174.5510396429664,
  ],
  Array [
    118.10962295627576,
    -188.5035185962929,
  ],
  Array [
    119.93268297910122,
    -197.11860503635728,
  ],
  Array [
    121.48494324921218,
    -198.42856927038895,
  ],
  Array [
    123.30559954409723,
    -199.32897412183223,
  ],
  Array [
    125.2888352343545,
    -199.7674880184159,
  ],
  Array [
    127.31938456848428,
    -199.71862451186698,
  ],
  Array [
    129.27923192378202,
    -199.18522355148195,
  ],
  Array [
    131.05447087987474,
    -198.19828642614019,
  ],
  Array [
    132.54192446522515,
    -196.81517396793754,
  ],
  Array [
    133.65514180756495,
    -195.11627273793,
  ],
  Array [
    134.32942266265303,
    -193.20032295542816,
  ],
  Array [
    134.52557779551077,
    -191.17867971181153,
  ],
  Array [
    134.23220666060013,
    -189.16884100738562,
  ],
  Array [
    133.4663600020661,
    -187.28761876208353,
  ],
  Array [
    132.27254886368584,
    -185.64434970119106,
  ],
  Array [
    131.22145893643386,
    -184.8843103079631,
  ],
  Array [
    125.96085426728841,
    -177.72755591680487,
  ],
  Array [
    120.40768139988137,
    -170.33331984830028,
  ],
  Array [
    113.85502001862912,
    -162.01926761239986,
  ],
  Array [
    105.74348271771476,
    -150.20198650864697,
  ],
  Array [
    101.20690262407587,
    -143.5511512200893,
  ],
  Array [
    92.07468191872637,
    -131.209216754576,
  ],
  Array [
    87.23161825975811,
    -123.94621473388507,
  ],
  Array [
    81.93061987199184,
    -115.82309404132988,
  ],
  Array [
    76.16395710517682,
    -107.07436211447894,
  ],
  Array [
    70.0420762822013,
    -97.54876738794232,
  ],
  Array [
    62.63357758405237,
    -85.48605106149196,
  ],
  Array [
    55.13315939141299,
    -72.58027978664668,
  ],
  Array [
    47.87115985980247,
    -60.379072960800826,
  ],
  Array [
    40.83369321196069,
    -48.31898984449831,
  ],
  Array [
    34.05286881600848,
    -37.418462815555,
  ],
  Array [
    27.373075439879276,
    -27.36912993802174,
  ],
  Array [
    21.11680350218738,
    -18.605627668028017,
  ],
  Array [
    10.13646067811521,
    -3.556971148710227,
  ],
  Array [
    4.382856963418741,
    3.5705125027592417,
  ],
  Array [
    3.4143777563645665,
    4.566380432591424,
  ],
  Array [
    2.2357083447341997,
    5.301528938848113,
  ],
  Array [
    0.9153530501111424,
    5.733231155804868,
  ],
  Array [
    -0.46994901574448544,
    5.836396530328719,
  ],
  Array [
    -1.839684031677017,
    5.6050290860051355,
  ],
  Array [
    -3.1142429342037286,
    5.052575909568958,
  ],
  Array [
    -4.219548300024874,
    4.2111456055915575,
  ],
  Array [
    -5.091359729346379,
    3.1296421428745345,
  ],
  Array [
    -5.679007500934192,
    1.8709225534852265,
  ],
  Array [
    -5.94833749869312,
    0.5081436790996228,
  ],
  Array [
    -5.883696250498996,
    -0.8794897080730903,
  ],
  Array [
    -5.48884070873402,
    -2.211328290372765,
  ],
  Array [
    -4.786719896037779,
    -3.4099655468199694,
  ],
]
`;

exports[`getStrokeOutlinePoints gets stroke outline points with a single point: get-stroke-outline-points-one-point 1`] = `
Array [
  Array [
    469.8101723753737,
    282.75982762462627,
  ],
  Array [
    470.9648255891402,
    284.36732752782905,
  ],
  Array [
    471.5457396171903,
    286.2593684562639,
  ],
  Array [
    471.49209260744067,
    288.23785322527357,
  ],
  Array [
    470.8095014159735,
    290.0956339572734,
  ],
  Array [
    469.5694335207221,
    291.63820051921647,
  ],
  Array [
    467.90172435766533,
    292.7040457848694,
  ],
  Array [
    465.9809835164078,
    293.1815754735625,
  ],
  Array [
    464.00831307005535,
    293.0207921008339,
  ],
  Array [
    462.19025213507564,
    292.2385297282511,
  ],
  Array [
    460.71715217128246,
    290.91669143272065,
  ],
  Array [
    459.7432471337456,
    289.19367403271605,
  ],
  Array [
    459.37050513698114,
    287.2498779048502,
  ],
  Array [
    459.63795236644467,
    285.28881901682297,
  ],
  Array [
    460.5175870266121,
    283.51582075188014,
  ],
  Array [
    461.9173111363629,
    282.11651649638725,
  ],
  Array [
    463.69057321191497,
    281.2374137752749,
  ],
  Array [
    465.65171224586214,
    280.9705548755041,
  ],
  Array [
    467.5953964636598,
    281.34387999432477,
  ],
  Array [
    469.31812161462176,
    282.31830189324813,
  ],
  Array [
    470.63951792068696,
    283.79179834223453,
  ],
  Array [
    471.4212348397956,
    285.61009387410974,
  ],
  Array [
    471.5814264041639,
    287.5828124667031,
  ],
  Array [
    471.10332051471596,
    289.50340996262287,
  ],
  Array [
    470.03697498428465,
    291.17079929705784,
  ],
  Array [
    468.49403647139405,
    292.4104043665446,
  ],
  Array [
    466.63605104564004,
    293.09243819308386,
  ],
  Array [
    464.65755027155956,
    293.14549165499756,
  ],
  Array [
    462.76568370247224,
    292.56401004081863,
  ],
  Array [
    461.15853026756594,
    291.4088746290478,
  ],
  Array [
    461.1598276246263,
    291.4101723753737,
  ],
  Array [
    460.09608311652863,
    290.3702702486252,
  ],
  Array [
    459.31212368387924,
    289.1060090336526,
  ],
  Array [
    458.8535130857379,
    287.6908676516711,
  ],
  Array [
    458.74690579161626,
    286.20709418629167,
  ],
  Array [
    458.99849782245764,
    284.740925621037,
  ],
  Array [
    459.5936666376085,
    283.37757574180716,
  ],
  Array [
    460.49782099756845,
    282.19628250763986,
  ],
  Array [
    461.65841140845504,
    281.26570273680767,
  ],
  Array [
    463.0079843010575,
    280.63992176931447,
  ],
  Array [
    464.46810243544826,
    280.3553100243932,
  ],
  Array [
    465.9539036770533,
    280.42840915003103,
  ],
  Array [
    467.37903318790035,
    280.85497062161136,
  ],
  Array [
    468.6606623738596,
    281.61020266636405,
  ],
]
`;

exports[`getStrokeOutlinePoints gets stroke outline points with duplicates: get-stroke-outline-points-duplicates 1`] = `
Array [
  Array [
    4.224688080042079,
    -4.224688080042079,
  ],
  Array [
    10.168344244742169,
    1.3316557552578319,
  ],
  Array [
    65.80475920006808,
    57.417924979619414,
  ],
  Array [
    87.50286525240143,
    79.86677552396576,
  ],
  Array [
    96.87876278474482,
    89.25333454521123,
  ],
  Array [
    102.96676306431297,
    94.52834701591033,
  ],
  Array [
    103.75529212050765,
    95.18012966358725,
  ],
  Array [
    104.65681228214318,
    96.33084424767203,
  ],
  Array [
    105.25674162717746,
    97.66387366851295,
  ],
  Array [
    105.52021223388954,
    99.10174219353863,
  ],
  Array [
    105.4319111782382,
    100.5608808365166,
  ],
  Array [
    104.99697052172671,
    101.95648439095966,
  ],
  Array [
    104.2406690357471,
    103.20744031271143,
  ],
  Array [
    103.2069629981992,
    104.24104298482376,
  ],
  Array [
    101.95593145255374,
    104.99721937142988,
  ],
  Array [
    100.56028441102313,
    105.43202046541147,
  ],
  Array [
    99.1011369452353,
    105.52017560675725,
  ],
  Array [
    97.66329477445959,
    105.25656121451026,
  ],
  Array [
    96.33032535321821,
    104.65649856953375,
  ],
  Array [
    95.17970092690302,
    103.75486334094761,
  ],
  Array [
    38.43703385226901,
    46.11052040597133,
  ],
  Array [
    14.406898944853781,
    21.525811614898366,
  ],
  Array [
    13.456662205181654,
    20.202898857344188,
  ],
  Array [
    12.978591520873394,
    18.64581939683893,
  ],
  Array [
    13.022741018774472,
    17.017599863926556,
  ],
  Array [
    13.584488234800315,
    15.488715273511204,
  ],
  Array [
    14.605018087146153,
    14.219240245795694,
  ],
  Array [
    15.977480830877656,
    13.342089151290704,
  ],
  Array [
    17.558179254431163,
    12.949099941136309,
  ],
  Array [
    19.181613815596755,
    13.08141868937475,
  ],
  Array [
    20.677810487275295,
    13.725191588136706,
  ],
  Array [
    21.890117082984474,
    14.813015444982257,
  ],
  Array [
    22.691604782135684,
    16.230994814942065,
  ],
  Array [
    22.99835761105558,
    17.83066688367114,
  ],
  Array [
    22.778258467581374,
    19.444545563301126,
  ],
  Array [
    22.054351790135343,
    20.90365733005217,
  ],
  Array [
    20.902430798938884,
    22.055232796777343,
  ],
  Array [
    19.443101925847895,
    22.77870170812411,
  ],
  Array [
    17.829157289100394,
    22.998316678097225,
  ],
  Array [
    16.229577318203866,
    22.691083961367784,
  ],
  Array [
    14.811838458359269,
    21.889170904478913,
  ],
  Array [
    13.724378342439053,
    20.676538016171378,
  ],
  Array [
    13.08105433164165,
    19.180148279954956,
  ],
  Array [
    12.949222619718594,
    17.556674096220046,
  ],
  Array [
    13.342686021708426,
    15.976093640559247,
  ],
  Array [
    14.220248815558557,
    14.603894103912967,
  ],
  Array [
    15.4900299450988,
    13.583745139993574,
  ],
  Array [
    17.01908299087662,
    13.0224566146166,
  ],
  Array [
    18.647315695368288,
    12.978795584554788,
  ],
  Array [
    20.204251664601827,
    13.457333371181011,
  ],
  Array [
    21.526879291607308,
    14.407966941913791,
  ],
  Array [
    21.525811614898366,
    14.406898944853781,
  ],
  Array [
    46.11052040597133,
    38.43703385226901,
  ],
  Array [
    103.75486334094761,
    95.17970092690302,
  ],
  Array [
    104.65649856953375,
    96.33032535321821,
  ],
  Array [
    105.25656121451026,
    97.66329477445959,
  ],
  Array [
    105.52017560675725,
    99.1011369452353,
  ],
  Array [
    105.43202046541147,
    100.56028441102313,
  ],
  Array [
    104.99721937142988,
    101.95593145255374,
  ],
  Array [
    104.24104298482376,
    103.2069629981992,
  ],
  Array [
    103.20744031271143,
    104.2406690357471,
  ],
  Array [
    101.95648439095966,
    104.99697052172671,
  ],
  Array [
    100.5608808365166,
    105.4319111782382,
  ],
  Array [
    99.10174219353863,
    105.52021223388954,
  ],
  Array [
    97.66387366851295,
    105.25674162717746,
  ],
  Array [
    96.33084424767203,
    104.65681228214318,
  ],
  Array [
    95.18012966358725,
    103.75529212050765,
  ],
  Array [
    94.52834701591033,
    102.96676306431297,
  ],
  Array [
    89.25333454521123,
    96.87876278474482,
  ],
  Array [
    79.86677552396576,
    87.50286525240143,
  ],
  Array [
    57.417924979619414,
    65.80475920006808,
  ],
  Array [
    1.3316557552578319,
    10.168344244742169,
  ],
  Array [
    -4.224688080042079,
    4.224688080042079,
  ],
  Array [
    -5.112984132720486,
    3.0908529576555086,
  ],
  Array [
    -5.704113307372647,
    1.7773773161468955,
  ],
  Array [
    -5.963719148593695,
    0.36060042151410965,
  ],
  Array [
    -5.876713352694334,
    -1.077134586559814,
  ],
  Array [
    -5.448152700620184,
    -2.452266481853931,
  ],
  Array [
    -4.702945157605631,
    -3.68487253400345,
  ],
  Array [
    -3.684402221064107,
    -4.7033136213436935,
  ],
  Array [
    -2.4517216543234426,
    -5.448397900027198,
  ],
  Array [
    -1.0765469098398524,
    -5.876821036769243,
  ],
  Array [
    0.3611967916249723,
    -5.963683058733008,
  ],
  Array [
    1.7779477185897927,
    -5.703935541120762,
  ],
  Array [
    3.091364240613661,
    -5.112675021860316,
  ],
  Array [
    4.225110527725936,
    -4.224265590111343,
  ],
]
`;

exports[`getStrokeOutlinePoints gets stroke outline points with two equal points: get-stroke-outline-points-two-equal-points 1`] = `
Array [
  Array [
    6.728352660193719,
    -0.7849301948399909,
  ],
  Array [
    6.901686161036018,
    2.081711817738876,
  ],
  Array [
    5.722930094936034,
    4.700531221101572,
  ],
  Array [
    3.462140126572477,
    6.471550602628259,
  ],
  Array [
    0.6372683201289022,
    6.989025440622029,
  ],
  Array [
    -2.1045008430072114,
    6.134401086374877,
  ],
  Array [
    -4.13502188385132,
    4.103473900706759,
  ],
  Array [
    -4.98909786717701,
    1.361533867536321,
  ],
  Array [
    -4.4710580651752165,
    -1.4632343874429035,
  ],
  Array [
    -2.6995865610782586,
    -3.7236701067167184,
  ],
  Array [
    -0.08053145888030677,
    -4.9019023853644,
  ],
  Array [
    2.786075829665783,
    -4.727995559590081,
  ],
  Array [
    5.243489130398231,
    -3.241792074134711,
  ],
]
`;

exports[`getStrokeOutlinePoints gets stroke outline points with two points: get-stroke-outline-points-two-points 1`] = `
Array [
  Array [
    4.893206982421875,
    200,
  ],
  Array [
    5.19757506225586,
    171.25,
  ],
  Array [
    5.4182419201354985,
    130.28125,
  ],
  Array [
    5.578225392098236,
    84.11953125,
  ],
  Array [
    5.694213409271221,
    35.75080078125,
  ],
  Array [
    5.919621905604509,
    34.375914323509875,
  ],
  Array [
    6.572247065536214,
    33.14497884082992,
  ],
  Array [
    7.583758862545174,
    32.186873603182136,
  ],
  Array [
    8.848251744471371,
    31.60191248186346,
  ],
  Array [
    10.233332972783426,
    31.451341060690705,
  ],
  Array [
    11.593984177251135,
    31.750924207207646,
  ],
  Array [
    12.787744817589694,
    32.469295486891696,
  ],
  Array [
    13.689627839633445,
    33.53124123712833,
  ],
  Array [
    14.205205852900416,
    34.82557545754132,
  ],
  Array [
    14.280497703302178,
    36.21678101358238,
  ],
  Array [
    13.907620314226547,
    37.55919831629139,
  ],
  Array [
    13.125614046739813,
    38.71227591971099,
  ],
  Array [
    12.016355163477442,
    39.55528629438021,
  ],
  Array [
    10.695983362332912,
    39.99996603153805,
  ],
  Array [
    9.302741919430359,
    39.999757045320045,
  ],
  Array [
    7.982503581621712,
    39.554681216638386,
  ],
  Array [
    6.8734976513846,
    38.711338102244646,
  ],
  Array [
    6.091837342363988,
    37.558025948836814,
  ],
  Array [
    5.719362695252612,
    36.215496843321525,
  ],
  Array [
    5.795071903926794,
    34.82431393743951,
  ],
  Array [
    6.3110381942530545,
    33.53013444867322,
  ],
  Array [
    7.213239759432362,
    32.4684593111267,
  ],
  Array [
    8.407215857432362,
    31.75044619195609,
  ],
  Array [
    9.767956875613374,
    31.451271254275607,
  ],
  Array [
    11.1529928701711,
    31.602258193034906,
  ],
  Array [
    12.417310206861348,
    32.187598635889216,
  ],
  Array [
    13.428534526785299,
    33.14600728395681,
  ],
  Array [
    14.08079037670961,
    34.37713849878971,
  ],
  Array [
    14.305786396968383,
    35.75209251720783,
  ],
  Array [
    14.30578659072878,
    35.75080078125,
  ],
  Array [
    14.421774607901764,
    84.11953125,
  ],
  Array [
    14.581758079864501,
    130.28125,
  ],
  Array [
    14.80242493774414,
    171.25,
  ],
  Array [
    15.106793017578125,
    200,
  ],
  Array [
    14.958389492516485,
    201.22217370487456,
  ],
  Array [
    14.521804137180853,
    202.37331461659824,
  ],
  Array [
    13.822411312936213,
    203.3865183712955,
  ],
  Array [
    12.900859763588496,
    204.2028975191382,
  ],
  Array [
    11.8107101084847,
    204.7750040656963,
  ],
  Array [
    10.615321896799927,
    205.06958715160349,
  ],
  Array [
    9.384171147562364,
    205.069525594066,
  ],
  Array [
    8.188812400163075,
    204.7748229708107,
  ],
  Array [
    7.098719961164591,
    204.20260741214784,
  ],
  Array [
    6.177250054339279,
    203.38613611323225,
  ],
  Array [
    5.4779585539669045,
    202.3728624243187,
  ],
  Array [
    5.04148831490518,
    201.22167785981526,
  ],
  Array [
    4.89320700795584,
    199.9994893206991,
  ],
]
`;
